### PR TITLE
feat: verifiable polymorphic, email views, html file download

### PR DESCRIPTION
**Pull request by thangtx-2276**
Change domain_verifications migration:
* Add verifiable_type
* Change verifiable_id to integer
* `activation_token`: token for verification by mail

Email views:
* Add default verification email views: resources/views/mail.blade.php
* Add route + view for successful + failed verification

HTML file download:
* Add route for getting html file
* Add `getHtmlFileUrl` method to any strategies for getting downloadable URL of html file
* Add `getDomainName` method to any stategies for getting domain name of URL

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,15 +5,16 @@
     "license": "MIT",
     "require": {
         "php": ">=7.1.3",
-        "illuminate/config": "~5.0|~5.7.0|~5.8.0|^6.0",
-        "illuminate/database": "~5.0|~5.7.0|~5.8.0|^6.0",
-        "illuminate/hashing": "^5.0",
-        "illuminate/support": "~5.0|~5.7.0|~5.8.0|^6.0",
+        "illuminate/config": "~5.0|~5.7.0|~5.8.0|^6.0|^7.0",
+        "illuminate/database": "~5.0|~5.7.0|~5.8.0|^6.0|^7.0",
+        "illuminate/hashing": "^5.0|^7.0",
```
Comment (kimyvgy) [inline composer.json]:
```suggestion
        "illuminate/hashing": "~5.0|~5.7.0|~5.8.0|^6.0|^7.0",
```

File: src/Supports/URL.php [hunk 2]
```diff
@@ -24,7 +24,6 @@ public static function normalize(string $url)
      */
     public static function getDomainName(string $url)
     {
-        $matched = preg_match('/(http[s]?:\/\/)(.+)/', $url, $data);
-        return $matched ? trim($data[2]) : '';
+        return parse_url($url)['host'];
```
Comment (kimyvgy) [inline src/Supports/URL.php]: Trường hợp parse lỗi thì sao nhỉ? :dagger:

File: src/Repositories/DomainVerification.php [hunk 3]
```diff
@@ -41,88 +42,32 @@ public function firstOrCreate(string $url, DomainVerifiableInterface $verifiable
...
-
     /**
      * Create a new randoom token
      *
      * @return string
      */
     protected function generateToken()
     {
-        return hash_hmac('sha256', str_random(48), $this->hashKey);
+        return Str::random(48);
```
Comment (kimyvgy) [inline src/Repositories/DomainVerification.php]:
```suggestion
        return hash_hmac('sha256', str_random(48), $this->hashKey);
```

File: src/Supports/URL.php (hunk 2 above)
Comment (thangtx-2276) [reply src/Supports/URL.php]: Em đã thêm đoạn kiểm tra & throw exception khi không thể tìm được host từ URL ạ

**Pull request merged**